lib/puppet/node.rb

```diff
@@ -39,7 +39,7 @@ def self.from_data_hash(data)
   def to_data_hash
     result = {
       'name' => name,
-      'environment' => environment.name,
+      'environment' => environment.name.to_s,
```

Comment:
We did miss those tests. In this PR, tests are added that asserts that everything returned from the to_data_hash methods is `Data`. The `Data` type does not accept symbols.